akka-http-core/src/main/scala/akka/http/scaladsl/model/ws/UpgradeToWebSocket.scala

```diff
@@ -88,6 +108,18 @@ trait UpgradeToWebSocket extends jm.ws.UpgradeToWebSocket {
     subprotocol: String): HttpResponse =
     handleMessages(createScalaFlow(inSink, outSource), subprotocol = Some(subprotocol))
 
+  /**
+   * Java API
+   */
+  def handleMessagesWithMat[Mat](handlerFlow: Graph[FlowShape[jm.ws.Message, jm.ws.Message], Mat]): (HttpResponse, Future[Mat]) =
+    handleMessagesWithMat(JavaMapping.toScala(handlerFlow))
+
+  /**
+   * Java API
+   */
+  def handleMessagesWithMat[Mat](handlerFlow: Graph[FlowShape[jm.ws.Message, jm.ws.Message], Mat], subprotocol: String): (HttpResponse, Future[Mat]) =
```

Comment:
this must return `akka.japi.Pair` instead, a Scala tuple is not usable from Java